notifications/source/client.go

```diff
@@ -0,0 +1,187 @@
+package source
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"iter"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/icinga/icinga-go-library/notifications/event"
+	"github.com/pkg/errors"
+)
+
+// ErrRulesOutdated implies that the rules version between Icinga DB and Icinga Notifications mismatches.
+var ErrRulesOutdated = fmt.Errorf("rules version is outdated")
+
+// BasicAuthTransport is an http.RoundTripper that adds basic authentication and a User-Agent header to HTTP requests.
+type BasicAuthTransport struct {
+	http.RoundTripper // RoundTripper is the underlying HTTP transport to use for making requests.
+
+	Username   string
+	Password   string
+	ClientName string // ClientName is used to set the User-Agent header.
+}
+
+// RoundTrip adds basic authentication headers to the request and executes the HTTP request.
+func (b *BasicAuthTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.SetBasicAuth(b.Username, b.Password)
+	// As long as our round tripper is used for the client, the User-Agent header below
+	// overrides any other value set by the user.
+	req.Header.Set("User-Agent", b.ClientName)
+
+	return b.RoundTripper.RoundTrip(req)
+}
+
+// Client provides a common interface to interact with the Icinga Notifications API.
+// It holds the configuration for the API endpoint and the HTTP client used to make requests.
+type Client struct {
+	cfg Config // cfg holds base API endpoint URL and authentication details.
+
+	client http.Client // HTTP client used for making requests to the Icinga Notifications API.
+
+	IcingaWebBaseUrl *url.URL // IcingaWebBaseUrl holds the base URL for Icinga Web 2.
+	Endpoints        struct {
+		ProcessEvent string // ProcessEvent holds the URL for the process event endpoint.
+	}
```

Comment:
I currently don't see a point in having that extra struct. I know that there was a second URL in there earlier, so it was probably intended for some grouping within the `Client` struct, but now it just looks strange.

Additionally, I don't see why this should be exported.

Reply:
Done.